iothub/device/tests/ConnectionString/IotHubConnectionStringBuilderTests.cs

```diff
@@ -0,0 +1,232 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using FluentAssertions;
+using Microsoft.Azure.Devices.Client;
+using Microsoft.Azure.Devices.Client.ApiTest;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Azure.Devices.Client.Test.ConnectionString
+{
+    [TestClass]
+    [TestCategory("Unit")]
+    public class IotHubConnectionStringBuilderTests
+    {
+        private const string HostName = "acme.azure-devices.net";
+        private const string GatewayHostName = "gateway.acme.azure-devices.net";
+        private const string TransparentGatewayHostName = "test";
+        private const string DeviceId = "device1";
+        private const string DeviceIdSplChar = "device1-.+%_#*?!(),=@;$'";
+        private const string ModuleId = "moduleId";
+        private const string SharedAccessKey = "dGVzdFN0cmluZzE=";
+        private const string SharedAccessKeyName = "AllAccessKey";
+        private const string CredentialScope = "Device";
+        private const string CredentialType = "SharedAccessSignature";
+        private const string SharedAccessSignature = "SharedAccessSignature sr=dh%3a%2f%2facme.azure-devices.net&sig=dGVzdFN0cmluZzU=&se=87824124985&skn=AllAccessKey";
+
+        [TestMethod]
+        public void IotHubConnectionStringBuilder_ParamConnectionString_ParsesHostName()
```

Comment:
I broke the one long test out into separate cases, so we more easily track what is being tested, and what the gaps may be.